### PR TITLE
Fix crash when import watch-only wallet and click send button

### LINKED ISSUE #182: Trying to import watch only wallet crashes the app
Note: This issue happened on a fresh installation of cryptopower (with no previous wallets)

``` 2023-10-13 15:49:03.542 [INF] WLLT: Upgrading database from version 1 to 25
2023-10-13 15:49:05.686 [INF] DCR: Opened wallet
2023-10-13 15:49:05.695 [INF] SHWL: Created Watching Only Wallet
panic: assignment to entry in nil map

goroutine 551 [running]:
github.com/crypto-power/cryptopower/libwallet/assets/dcr.(*Asset).AddAccountMixerNotificationListener(0x14002a6c340, {0x1024369e0?, 0x1400277f000}, {0x101943021, 0x8})
	/Users/kennedy/go/src/github.com/crypto-power/cryptopower/libwallet/assets/dcr/account_mixer.go:36 +0xb8
github.com/crypto-power/cryptopower/ui/page/root.(*OverviewPage).listenForMixerNotifications(0x1400277f000)
	/Users/kennedy/go/src/github.com/crypto-power/cryptopower/ui/page/root/overview_page.go:999 +0x1b0
github.com/crypto-power/cryptopower/ui/page/root.(*OverviewPage).OnNavigatedTo(0x1400277f000)
	/Users/kennedy/go/src/github.com/crypto-power/cryptopower/ui/page/root/overview_page.go:198 +0x14c
github.com/crypto-power/cryptopower/app.(*PageStack).Push(0x14003008660, {0x102440160, 0x1400277f000}, {0x102443770, 0x14002f82070})
	/Users/kennedy/go/src/github.com/crypto-power/cryptopower/app/pagestack.go:67 +0x460
github.com/crypto-power/cryptopower/app.(*MasterPage).Display(0x14002f82070, {0x102440160?, 0x1400277f000?})
	/Users/kennedy/go/src/github.com/crypto-power/cryptopower/app/masterpage.go:46 +0x3c
github.com/crypto-power/cryptopower/ui/page/root.(*HomePage).OnNavigatedTo(0x14002e0e000)
	/Users/kennedy/go/src/github.com/crypto-power/cryptopower/ui/page/root/home_page.go:138 +0xec
github.com/crypto-power/cryptopower/app.(*PageStack).Push(0x14002b063f0, {0x10243f308, 0x14002e0e000}, {0x102443720, 0x14002b06420})
	/Users/kennedy/go/src/github.com/crypto-power/cryptopower/app/pagestack.go:67 +0x460
github.com/crypto-power/cryptopower/app.(*SimpleWindowNavigator).Display(0x14002b06420, {0x10243f308?, 0x14002e0e000?})
	/Users/kennedy/go/src/github.com/crypto-power/cryptopower/app/window.go:49 +0x3c
github.com/crypto-power/cryptopower/ui/page/root.(*CreateWallet).walletCreationSuccessCallback(0x14002a2a850?)
	/Users/kennedy/go/src/github.com/crypto-power/cryptopower/ui/page/root/wallet_setup_page.go:630 +0x54
github.com/crypto-power/cryptopower/ui/page/root.(*CreateWallet).HandleUserInteractions.func3()
	/Users/kennedy/go/src/github.com/crypto-power/cryptopower/ui/page/root/wallet_setup_page.go:551 +0x490
created by github.com/crypto-power/cryptopower/ui/page/root.(*CreateWallet).HandleUserInteractions in goroutine 70
	/Users/kennedy/go/src/github.com/crypto-power/cryptopower/ui/page/root/wallet_setup_page.go:514 +0x5e4
joyceofoche@Joyces-MacBook-Air ~ % ```

### PR DESCRIPTION
**Pull request by JustinBeBoy**
Resolve #182 

This PR fix:
- App crash when import only watch-only wallet.
- App crash when click send button
- App crash when import mixed account of watch-only wallet
- App crash when delete wallet.

pr1

### CODE REVIEW

**Comment by dreacot** [comment]
launching a fresh instance of cryptopower i get this error

```
$ go run . --network=testnet
Config file (/home/kennedy/.cryptopower/cryptopower.conf) does not exist. Using defaults.
2023-10-24 11:08:25.572 [INF] DLWL: Starting check and remove all dir of deleted wallets....
2023-10-24 11:08:25.572 [ERR] DLWL: can't read DCR root wallet type: open /home/kennedy/.cryptopower/testnet/testnet3/dcr: no such file or directory
2023-10-24 11:08:25.572 [INF] DLWL: Loaded 0 wallets
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x10de1ce]

goroutine 1 [running]:
github.com/crypto-power/cryptopower/libwallet.(*AssetsManager).GetCurrencyConversionExchange(...)
        /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/libwallet/assets_config.go:141
github.com/crypto-power/cryptopower/libwallet.(*AssetsManager).initRateSource(0xc000511780)
        /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/libwallet/assets_manager.go:193 +0x10e
github.com/crypto-power/cryptopower/libwallet.NewAssetsManager({0xc00050f780?, 0x7?}, {0x186c689, 0x3}, {0x18a5e86, 0x25}, {0xc000446d20, 0x28}, {0x1871ea3, 0x7})
        /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/libwallet/assets_manager.go:172 +0x6c5
github.com/crypto-power/cryptopower/wallet.(*Wallet).InitAssetsManager(0xc000511700)
        /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/wallet/wallet.go:81 +0xbb
main.main()
        /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/main.go:77 +0x325
exit status 2
```

**Comment by dreacot** [comment]
> launching a fresh instance of cryptopower i get this error
> 
> ```
> $ go run . --network=testnet
> Config file (/home/kennedy/.cryptopower/cryptopower.conf) does not exist. Using defaults.
> 2023-10-24 11:08:25.572 [INF] DLWL: Starting check and remove all dir of deleted wallets....
> 2023-10-24 11:08:25.572 [ERR] DLWL: can't read DCR root wallet type: open /home/kennedy/.cryptopower/testnet/testnet3/dcr: no such file or directory
> 2023-10-24 11:08:25.572 [INF] DLWL: Loaded 0 wallets
> panic: runtime error: invalid memory address or nil pointer dereference
> [signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x10de1ce]
> 
> goroutine 1 [running]:
> github.com/crypto-power/cryptopower/libwallet.(*AssetsManager).GetCurrencyConversionExchange(...)
>         /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/libwallet/assets_config.go:141
> github.com/crypto-power/cryptopower/libwallet.(*AssetsManager).initRateSource(0xc000511780)
>         /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/libwallet/assets_manager.go:193 +0x10e
> github.com/crypto-power/cryptopower/libwallet.NewAssetsManager({0xc00050f780?, 0x7?}, {0x186c689, 0x3}, {0x18a5e86, 0x25}, {0xc000446d20, 0x28}, {0x1871ea3, 0x7})
>         /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/libwallet/assets_manager.go:172 +0x6c5
> github.com/crypto-power/cryptopower/wallet.(*Wallet).InitAssetsManager(0xc000511700)
>         /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/wallet/wallet.go:81 +0xbb
> main.main()
>         /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/main.go:77 +0x325
> exit status 2
> ```

this seems to be on master as well, but this would need to be fixed in order to test this PR as i need to import a watch only wallet on a fresh install of cryptopower

**Comment by JustinBeBoy** [comment]
> > launching a fresh instance of cryptopower i get this error
> > ```
> > $ go run . --network=testnet
> > Config file (/home/kennedy/.cryptopower/cryptopower.conf) does not exist. Using defaults.
> > 2023-10-24 11:08:25.572 [INF] DLWL: Starting check and remove all dir of deleted wallets....
> > 2023-10-24 11:08:25.572 [ERR] DLWL: can't read DCR root wallet type: open /home/kennedy/.cryptopower/testnet/testnet3/dcr: no such file or directory
> > 2023-10-24 11:08:25.572 [INF] DLWL: Loaded 0 wallets
> > panic: runtime error: invalid memory address or nil pointer dereference
> > [signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x10de1ce]
> > 
> > goroutine 1 [running]:
> > github.com/crypto-power/cryptopower/libwallet.(*AssetsManager).GetCurrencyConversionExchange(...)
> >         /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/libwallet/assets_config.go:141
> > github.com/crypto-power/cryptopower/libwallet.(*AssetsManager).initRateSource(0xc000511780)
> >         /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/libwallet/assets_manager.go:193 +0x10e
> > github.com/crypto-power/cryptopower/libwallet.NewAssetsManager({0xc00050f780?, 0x7?}, {0x186c689, 0x3}, {0x18a5e86, 0x25}, {0xc000446d20, 0x28}, {0x1871ea3, 0x7})
> >         /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/libwallet/assets_manager.go:172 +0x6c5
> > github.com/crypto-power/cryptopower/wallet.(*Wallet).InitAssetsManager(0xc000511700)
> >         /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/wallet/wallet.go:81 +0xbb
> > main.main()
> >         /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/main.go:77 +0x325
> > exit status 2
> > ```
> 
> this seems to be on master as well, but this would need to be fixed in order to test this PR as i need to import a watch only wallet on a fresh install of cryptopower

even though it's not within the scope of this PR, I'll fix it

**Comment by dreacot** [comment]
seems a PR has been opened

**Comment by dreacot** [comment]
@JustinBeBoy did you test this on a fresh install of cryptopower? cause that's how to detect the issue, it still happens

```
2023-10-24 17:38:42.444 [INF] DLWL: Loaded 0 wallets
2023-10-24 17:38:58.243 [INF] WLLT: Upgrading database from version 1 to 25
2023-10-24 17:39:00.939 [INF] DCR: Opened wallet
2023-10-24 17:39:00.945 [INF] SHWL: Created Watching Only Wallet
panic: assignment to entry in nil map

goroutine 39 [running]:
github.com/crypto-power/cryptopower/libwallet/assets/dcr.(*Asset).AddAccountMixerNotificationListener(0xc000648680, {0x1ff5578?, 0xc0000d4800}, {0x1877f8b, 0x8})
        /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/libwallet/assets/dcr/account_mixer.go:36 +0xf0
github.com/crypto-power/cryptopower/ui/page/root.(*OverviewPage).listenForMixerNotifications(0xc0000d4800)
        /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/ui/page/root/overview_page.go:1187 +0x22c
github.com/crypto-power/cryptopower/ui/page/root.(*OverviewPage).OnNavigatedTo(0xc0000d4800)
        /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/ui/page/root/overview_page.go:217 +0x22f
github.com/crypto-power/cryptopower/app.(*PageStack).Push(0xc003742d20, {0x1ffe6d8, 0xc0000d4800}, {0x2001700, 0xc000554090})
        /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/app/pagestack.go:67 +0x538
github.com/crypto-power/cryptopower/app.(*MasterPage).Display(0xc000554090, {0x1ffe6d8?, 0xc0000d4800?})
        /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/app/masterpage.go:46 +0x39
github.com/crypto-power/cryptopower/ui/page/root.(*HomePage).OnNavigatedTo(0xc000628000)
        /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/ui/page/root/home_page.go:137 +0x109
github.com/crypto-power/cryptopower/app.(*PageStack).Push(0xc0001c99e0, {0x1ffe648, 0xc000628000}, {0x2001750, 0xc0001c9a10})
        /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/app/pagestack.go:67 +0x538
github.com/crypto-power/cryptopower/app.(*SimpleWindowNavigator).Display(0xc0001c9a10, {0x1ffe648?, 0xc000628000?})
        /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/app/window.go:49 +0x39
github.com/crypto-power/cryptopower/ui/page/root.(*CreateWallet).walletCreationSuccessCallback(0xc00157d500)
        /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/ui/page/root/wallet_setup_page.go:636 +0x7a
github.com/crypto-power/cryptopower/ui/page/root.(*CreateWallet).HandleUserInteractions.func3()
        /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/ui/page/root/wallet_setup_page.go:554 +0x518
created by github.com/crypto-power/cryptopower/ui/page/root.(*CreateWallet).HandleUserInteractions
        /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/ui/page/root/wallet_setup_page.go:517 +0x745
exit status 2
```

**Comment by JustinBeBoy** [comment]
> @JustinBeBoy did you test this on a fresh install of cryptopower? cause that's how to detect the issue, it still happens
> 
> ```
> 2023-10-24 17:38:42.444 [INF] DLWL: Loaded 0 wallets
> 2023-10-24 17:38:58.243 [INF] WLLT: Upgrading database from version 1 to 25
> 2023-10-24 17:39:00.939 [INF] DCR: Opened wallet
> 2023-10-24 17:39:00.945 [INF] SHWL: Created Watching Only Wallet
> panic: assignment to entry in nil map
> 
> goroutine 39 [running]:
> github.com/crypto-power/cryptopower/libwallet/assets/dcr.(*Asset).AddAccountMixerNotificationListener(0xc000648680, {0x1ff5578?, 0xc0000d4800}, {0x1877f8b, 0x8})
>         /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/libwallet/assets/dcr/account_mixer.go:36 +0xf0
> github.com/crypto-power/cryptopower/ui/page/root.(*OverviewPage).listenForMixerNotifications(0xc0000d4800)
>         /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/ui/page/root/overview_page.go:1187 +0x22c
> github.com/crypto-power/cryptopower/ui/page/root.(*OverviewPage).OnNavigatedTo(0xc0000d4800)
>         /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/ui/page/root/overview_page.go:217 +0x22f
> github.com/crypto-power/cryptopower/app.(*PageStack).Push(0xc003742d20, {0x1ffe6d8, 0xc0000d4800}, {0x2001700, 0xc000554090})
>         /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/app/pagestack.go:67 +0x538
> github.com/crypto-power/cryptopower/app.(*MasterPage).Display(0xc000554090, {0x1ffe6d8?, 0xc0000d4800?})
>         /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/app/masterpage.go:46 +0x39
> github.com/crypto-power/cryptopower/ui/page/root.(*HomePage).OnNavigatedTo(0xc000628000)
>         /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/ui/page/root/home_page.go:137 +0x109
> github.com/crypto-power/cryptopower/app.(*PageStack).Push(0xc0001c99e0, {0x1ffe648, 0xc000628000}, {0x2001750, 0xc0001c9a10})
>         /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/app/pagestack.go:67 +0x538
> github.com/crypto-power/cryptopower/app.(*SimpleWindowNavigator).Display(0xc0001c9a10, {0x1ffe648?, 0xc000628000?})
>         /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/app/window.go:49 +0x39
> github.com/crypto-power/cryptopower/ui/page/root.(*CreateWallet).walletCreationSuccessCallback(0xc00157d500)
>         /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/ui/page/root/wallet_setup_page.go:636 +0x7a
> github.com/crypto-power/cryptopower/ui/page/root.(*CreateWallet).HandleUserInteractions.func3()
>         /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/ui/page/root/wallet_setup_page.go:554 +0x518
> created by github.com/crypto-power/cryptopower/ui/page/root.(*CreateWallet).HandleUserInteractions
>         /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/ui/page/root/wallet_setup_page.go:517 +0x745
> exit status 2
> ```

I have tested, seem like your wallet is the have mixer, I will test that case

**Comment by dreacot** [comment]
clicking the send button for the first time caused a crash
however opening the app again and clicking the send button shows a modal.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x60 pc=0x12eb8b4]

goroutine 10 [running]:
github.com/crypto-power/cryptopower/ui/page/send.(*Page).showBalanceAfterSend(0xc0000be300?)
        /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/ui/page/send/page.go:455 +0x34
github.com/crypto-power/cryptopower/ui/page/send.(*Page).validateAndConstructTx(0x0?)
        /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/ui/page/send/page.go:341 +0x65
github.com/crypto-power/cryptopower/ui/page/send.NewSendPage.func2()
        /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/ui/page/send/page.go:138 +0x1d
github.com/crypto-power/cryptopower/ui/page/send.(*destination).handle(0xc0050d1c00)
        /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/ui/page/send/send_destination.go:142 +0x65
github.com/crypto-power/cryptopower/ui/page/send.(*Page).HandleUserInteractions(0xc0000be300)
        /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/ui/page/send/page.go:510 +0x2bd
github.com/crypto-power/cryptopower/ui/page/send.(*Page).Handle(0xc000478a00?)
        /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/ui/page/send/page.go:603 +0xbb
github.com/crypto-power/cryptopower/ui.(*Window).handleFrameEvent(0xc000191680, {{0xc146710844bbd07f, 0x126d34640b, 0x2f08680}, {0x3f800000, 0x3f800000}, {0x320, 0x28a}, {0x0, 0x0, ...}, ...})
        /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/ui/window.go:235 +0x171
github.com/crypto-power/cryptopower/ui.(*Window).HandleEvents(0xc000191680)
        /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/ui/window.go:206 +0x2b9
main.main.func3()
        /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/main.go:110 +0x1d
created by main.main
        /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/main.go:108 +0x61b
exit status 2
```

File: app/masterpage.go [hunk 1]
```diff
@@ -85,6 +87,8 @@ func (masterPage *MasterPage) ClearStackAndDisplay(newPage Page) {
 // CloseAllPages dismisses all pages in the stack.
 // Part of the PageNavigator interface.
 func (masterPage *MasterPage) CloseAllPages() {
+	fmt.Println("------CloseAllPages---111----")
```
Comment (dreacot) [inline app/masterpage.go:90]: remove print statement

Comment (dreacot) [inline app/masterpage.go:90]: remove print statement

File: app/masterpage.go [hunk 2]
```diff
@@ -85,6 +87,8 @@ func (masterPage *MasterPage) ClearStackAndDisplay(newPage Page) {
...
 	masterPage.subPages.Reset()
+	fmt.Println("------CloseAllPages---2222----")
```
Comment (dreacot) [inline app/masterpage.go:92]: remove print statement

**Review by dreacot — CHANGES_REQUESTED** [review]

**Comment by JustinBeBoy** [comment]
> clicking the send button for the first time caused a crash however opening the app again and clicking the send button shows a modal.
> 
> ```
> panic: runtime error: invalid memory address or nil pointer dereference
> [signal SIGSEGV: segmentation violation code=0x1 addr=0x60 pc=0x12eb8b4]
> 
> goroutine 10 [running]:
> github.com/crypto-power/cryptopower/ui/page/send.(*Page).showBalanceAfterSend(0xc0000be300?)
>         /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/ui/page/send/page.go:455 +0x34
> github.com/crypto-power/cryptopower/ui/page/send.(*Page).validateAndConstructTx(0x0?)
>         /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/ui/page/send/page.go:341 +0x65
> github.com/crypto-power/cryptopower/ui/page/send.NewSendPage.func2()
>         /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/ui/page/send/page.go:138 +0x1d
> github.com/crypto-power/cryptopower/ui/page/send.(*destination).handle(0xc0050d1c00)
>         /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/ui/page/send/send_destination.go:142 +0x65
> github.com/crypto-power/cryptopower/ui/page/send.(*Page).HandleUserInteractions(0xc0000be300)
>         /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/ui/page/send/page.go:510 +0x2bd
> github.com/crypto-power/cryptopower/ui/page/send.(*Page).Handle(0xc000478a00?)
>         /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/ui/page/send/page.go:603 +0xbb
> github.com/crypto-power/cryptopower/ui.(*Window).handleFrameEvent(0xc000191680, {{0xc146710844bbd07f, 0x126d34640b, 0x2f08680}, {0x3f800000, 0x3f800000}, {0x320, 0x28a}, {0x0, 0x0, ...}, ...})
>         /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/ui/window.go:235 +0x171
> github.com/crypto-power/cryptopower/ui.(*Window).HandleEvents(0xc000191680)
>         /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/ui/window.go:206 +0x2b9
> main.main.func3()
>         /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/main.go:110 +0x1d
> created by main.main
>         /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/main.go:108 +0x61b
> exit status 2
> ```

I can't reproduce it, please explain step by step and the extended public key you use

**Comment by JustinBeBoy** [comment]
> > clicking the send button for the first time caused a crash however opening the app again and clicking the send button shows a modal.
> > ```
> > panic: runtime error: invalid memory address or nil pointer dereference
> > [signal SIGSEGV: segmentation violation code=0x1 addr=0x60 pc=0x12eb8b4]
> > 
> > goroutine 10 [running]:
> > github.com/crypto-power/cryptopower/ui/page/send.(*Page).showBalanceAfterSend(0xc0000be300?)
> >         /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/ui/page/send/page.go:455 +0x34
> > github.com/crypto-power/cryptopower/ui/page/send.(*Page).validateAndConstructTx(0x0?)
> >         /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/ui/page/send/page.go:341 +0x65
> > github.com/crypto-power/cryptopower/ui/page/send.NewSendPage.func2()
> >         /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/ui/page/send/page.go:138 +0x1d
> > github.com/crypto-power/cryptopower/ui/page/send.(*destination).handle(0xc0050d1c00)
> >         /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/ui/page/send/send_destination.go:142 +0x65
> > github.com/crypto-power/cryptopower/ui/page/send.(*Page).HandleUserInteractions(0xc0000be300)
> >         /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/ui/page/send/page.go:510 +0x2bd
> > github.com/crypto-power/cryptopower/ui/page/send.(*Page).Handle(0xc000478a00?)
> >         /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/ui/page/send/page.go:603 +0xbb
> > github.com/crypto-power/cryptopower/ui.(*Window).handleFrameEvent(0xc000191680, {{0xc146710844bbd07f, 0x126d34640b, 0x2f08680}, {0x3f800000, 0x3f800000}, {0x320, 0x28a}, {0x0, 0x0, ...}, ...})
> >         /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/ui/window.go:235 +0x171
> > github.com/crypto-power/cryptopower/ui.(*Window).HandleEvents(0xc000191680)
> >         /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/ui/window.go:206 +0x2b9
> > main.main.func3()
> >         /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/main.go:110 +0x1d
> > created by main.main
> >         /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/main.go:108 +0x61b
> > exit status 2
> > ```
> 
> I can't reproduce it, please explain step by step and the extended public key you use

After a lot of test, I have found the problem

File: ui/cryptomaterial/slider.go [hunk 3]
```diff
@@ -57,8 +57,12 @@ func (s *Slider) Layout(gtx C, items []layout.Widget) D {
 		s.items = items
 		s.isSliderItemsSet = true
 	}
-	s.handleClickEvent()
 
+	if len(s.items) == 0 {
+		return D{}
+	}
```
Comment (dreacot) [inline ui/cryptomaterial/slider.go:63]: i see no harm in showing the asset slider for a watch only wallet

File: ui/values/localizable/en.go [hunk 4]
```diff
@@ -787,4 +787,5 @@ const EN = `
 "bondStrengthErrMsg" = "Bond Strength must be a valid number"
 "minimumBondStrength" = "Minimum Bond Strength is %d"
 "assets" = "Assets"
+"noWalletsAvailable" = "There are no wallets available for this function, Please try again later."
```
Comment (dreacot) [inline ui/values/localizable/en.go:790]: telling them to try again later might not be of much help

we know the reason why they can't access the send page, so we should let them know

e.g. "You cannot spend from a watch only wallet, try creating another wallet"

**Review by dreacot — CHANGES_REQUESTED** [review]

File: ui/cryptomaterial/slider.go (hunk 3 above)
Comment (dreacot) [reply ui/cryptomaterial/slider.go:63]: or is a watch only wallet's balance not calculated as part of the slider?

if it's not then ignore this

Comment (JustinBeBoy) [reply ui/cryptomaterial/slider.go:63]: > or is a watch only wallet's balance not calculated as part of the slider?
> 
> if it's not then ignore this

Exactly, calculate total assets balance not include watch-only wallet, additional it causes crash app

**Review by dreacot — APPROVED** [review]

**Pull request merged**